src/app/api/services/dot-push-publish-dialog/dot-push-publish-dialog.service.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { PushPublishEvent } from '@models/push-publish-data/push-publish-data';
+
+@Injectable()
+export class DotPushPublishDialogService {
+    private _showDialog: Subject<PushPublishEvent> = new Subject<PushPublishEvent>();
+
+    get showDialog$(): Observable<PushPublishEvent> {
+        return this._showDialog.asObservable();
+    }
+
+    openDialog(data: PushPublishEvent): void {
```

Comment:
```suggestion
    open(data: PushPublishEvent): void {
```

We already have the context.